core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -486,6 +504,7 @@ private[spark] class BlockManagerInfo(
           blockId, blockManagerId.hostPort,
           Utils.bytesToString(blockStatus.externalBlockStoreSize)))
       }
+      if (!blockId.isBroadcast && blockStatus.isCached) _cachedBlocks += blockId
```

Comment:
Right, I guess I was a bit sleepy at this point. Not sure how I can test this case of a block being removed - simply unpersisting the RDD and then causing memory pressure on the executors should work I guess?